apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,61 +1,48 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgStyle } from '@angular/common';
+import { Component, input, output } from '@angular/core';
+import { HighlightDirective } from '../../shared/directives/highlight.directive';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   template: `
     <div
       class="flex w-fit flex-col gap-3 rounded-md border-2 border-black p-4"
-      [class]="customClass">
-      <img
-        *ngIf="type === CardType.TEACHER"
-        src="assets/img/teacher.png"
-        width="200px" />
-      <img
-        *ngIf="type === CardType.STUDENT"
-        src="assets/img/student.webp"
-        width="200px" />
-
+      [bgColor]="backgroundColor()">
+      <div class="card__image">
+        <ng-content select="[image]"></ng-content>
+      </div>
       <section>
-        <app-list-item
-          *ngFor="let item of list"
-          [name]="item.firstName"
-          [id]="item.id"
-          [type]="type"></app-list-item>
+        @for (item of list(); track item.id) {
+          <app-list-item [name]="item.firstName || item.name || ''">
+            <button (click)="deleteItem.emit(item.id)" delete>
+              <img class="h-5" src="assets/svg/trash.svg" />
+            </button>
+          </app-list-item>
+        }
       </section>
 
       <button
         class="rounded-sm border border-blue-500 bg-blue-300 p-2"
-        (click)="addNewItem()">
+        (click)="this.addNewItem.emit()">
         Add
       </button>
     </div>
   `,
+  styles: `
+    .card__image {
+      width: 200px;
```

Comment:
Why do we need to hard-code the width of the image here, if our goal is to have an abstract card-implementation?

`#api` `#no-blocker`

Reply:
Indeed, put this back to the specific card as it was initially.